core/src/services/hdfs_native/writer.rs

```diff
@@ -18,25 +18,29 @@
 use hdfs_native::file::FileWriter;
 
 use crate::raw::oio;
+use crate::services::hdfs_native::error::parse_hdfs_error;
 use crate::*;
 
 pub struct HdfsNativeWriter {
-    _f: FileWriter,
+    f: FileWriter,
 }
 
 impl HdfsNativeWriter {
     pub fn new(f: FileWriter) -> Self {
-        HdfsNativeWriter { _f: f }
+        HdfsNativeWriter { f }
     }
 }
 
 impl oio::Write for HdfsNativeWriter {
-    async fn write(&mut self, _bs: Buffer) -> Result<usize> {
-        todo!()
+    async fn write(&mut self, bs: Buffer) -> Result<usize> {
+        let bytes = bs.to_bytes();
+        let total_bytes = bytes.len();
+        self.f.write(bytes).await.map_err(parse_hdfs_error)?;
+        Ok(total_bytes)
```

Comment:
Does `self.f.write()` make sure that all bytes are written?

Reply:
Yes, based on the definition, it continues to write until completer `buf` has been written.

```
    pub async fn write(&mut self, mut buf: Bytes) -> Result<usize> {
        let bytes_to_write = buf.len();
        // Create a shallow copy of the bytes instance to mutate and track what's been read
        while !buf.is_empty() {
            let block_writer = self.get_block_writer().await?;

            block_writer.write(&mut buf).await?;
        }

        self.bytes_written += bytes_to_write;

        Ok(bytes_to_write)
    }
```

Reply:
Please don't rely this behavior. Return written like this:

```rust
let n = self.f.write(bytes).await.map_err(parse_hdfs_error)?;
Ok(n)
```